### PR TITLE
Union indexers in object types

### LINKED ISSUE #108: Unions as Obejct indexers
First, thank you for publishing this tool, my team is very excited to make the switch. We all appreciate your efforts.

When using aliases or union types, the translation as an object indexer doesn't work. [Example](https://flow-to-ts.netlify.com/#code=Ly8gQGZsb3cKdHlwZSBNeUN1c3RvbUtleSA9IHN0cmluZyB8IG51bWJlcgoKdHlwZSBNeUN1c3RvbU9iamVjdCA9IHsKCVtNeUN1c3RvbUtleV06IG51bWJlciwKfQoKdHlwZSBEYXkgPSAid2Vla2RheSIgfCAid2Vla2VuZCI7Cgp0eXBlIERheVN0YXRzID0gewogICAgW0RheV06IG51bWJlcgp9&prettier=1&semi=1&singleQuote=0&tabWidth=4&trailingComma=all&bracketSpacing=0&arrowParens=avoid&printWidth=80&inlineUtilityTypes=0)

Current output: 

```typescript
type DayStats = {
    [key: Day]: number;
};
```

This [throws errors](http://www.typescriptlang.org/play/#code/C4TwDgpgBAsiDCBXAzsA9gWwNIRFAvFKgE4CWAdgOZQA+U5iGARhMQNwBQHoksCK6DAHkmAKwgBjYASgBvDlEVQA2gGtcALj5JUmHCAC6Whs1acAvp27hoAEQCGeQgCIA7hAiqAJo+e0obh7q5F7OVjx2jgDKwPbAyDLySirqIFoOhsaMLOwcllwSaOSoUD4gMXHI6dGx8YkKSu6eZVoADAA0DYpNwV5aAIyteUA) with both aliases and union types,

Expected output:

```typescript
type DayStats = {
    [key in Day]: number;
};
```

Or even better:

```typescript
type DayStats = Record<Day, number>;
```

I've been playing with the `ObjectTypeIndexer` transformation, but I'm guessing it shouldn't be a `TSTypeLiteral` in the first place.

According to [this example](https://astexplorer.net/#/gist/b70a899ee34c67cba3050271c352daa9/58b1f97178b34f2ea264a466ed888409f245453f), I should be looking for a `TSMappedType`.

Any pointers?

### PR DESCRIPTION
**Pull request by lautis**
Allow using unions and enums as indexers in object types. The PR converts all object types that are not TypeScript keyswords (any, string, etc) in key to a mapped type.

Fixes #108.

Instead of

```
type DayStats = {
    [key: Day]: number;
};
```

the converter now outputs:

```
type DayStats = { [key in Day]?: number; };
```

The output does not retain newlines. More work would be required to split the object definition to multiple lines.

### CODE REVIEW

**Comment by k4b7** [comment]
Even though this is a new PR since we moved to Github Actions for running tests it looks like the workflow isn't running on this PR even though it ran on #120.  My guess is that Github Actions needs to be enable on your fork of this repo.  Do you have Github Actions enabled?

**Comment by lautis** [comment]
Yes, and GitHub Actions did run the workflow on my fork: https://github.com/lautis/flow-to-ts/runs/267411580

**Comment by lautis** [comment]
If I had to guess, it might be that the action also needs to define `on: pull_request`.

**Comment by k4b7** [comment]
That makes a lot of sense.

**Comment by k4b7** [comment]
I changed `workflows/node.yml` to include `on: [push, pull_request]`, but now the build fails with:
```
### ERRORED 21:53:13Z

- Only actions in "Khan" are allowed for this repository
```

**Comment by k4b7** [comment]
Closing and re-opening has seemed to fix the `Node CI` workflow.  I'm not sure what's up with the other `/ Build Failed`.

**Comment by k4b7** [comment]
The good news is the tests are running now.  For some reason the CODECOV_TOKEN secret is being injected for PRs that I make but not other PRs.

**Comment by joebowbeer** [comment]
@kevinbarabash See Settings > Actions

You probably do not have the first (most permissive) option checked.

**Comment by k4b7** [comment]
@joebowbeer I have `Enable local & third party Actions for this repository` checked in the actions setting. I think secrets are handled differently since a malicious PR could include a command that could POST the secret to some service to capture the secret.  I'm going to post an issue in the repo for the codecov-action.

**Comment by k4b7** [comment]
https://github.com/codecov/codecov-action/issues/29

**Comment by k4b7** [comment]
I ended up having to switch back to circleci to run codecov.  There doesn't currently seem to be a way to run codecov using github actions on PRs from forks without leaking API keys.

**Comment by k4b7** [comment]
I'll review this PR this weekend.  Sorry for the delay.

File: src/transform.js [hunk 1]
```diff
@@ -551,7 +551,29 @@ const transform = {
...
+          elements.push(indexer);
+        } else {
+          const typeParameter = t.tsTypeParameter(key);
+          typeParameter.name = indexer.parameters[0].name;
+
+          const mappedType = {
+            type: "TSMappedType",
+            typeParameter: typeParameter,
+            typeAnnotation: value,
+            optional: true
```
Comment (k4b7) [inline src/transform.js:571]: I think we should be copying the optionality of the flow type.

Comment (lautis) [reply src/transform.js:571]: The behaviour is different between Flow and TypeScript:

```
type Day = 'monday' | 'tuesday';

type DayStats = {
    [key: Day]: number;
};


const a: DayStats = {
  monday: 1
}
```

is ok for flow, but TypeScript complains about `a` missing key tuesday. What would be the Flow syntax for enforcing all keys in enum?

Comment (k4b7) [reply src/transform.js:571]: What you have makes sense.  Thanks for the explanation.

**Review by k4b7 — APPROVED** [review]
LGTM.  Thanks for adding tests.

**Pull request merged**